### PR TITLE
Migrated separate x, y coordinates to a Point class

### PR DESCRIPTION
**Pull request by Nohus**
This removes a lot of boilerplate,100 lines of code removed for the same functionality. Calls become cleaner, argument lists smaller, and less variables are necessary, with less opportunity for errors.

The potential issue with this would be additional allocations for the `Point` class, which I already looked into solving in the PR in which I added the class, by making it a value class. In the end I found such a value class already exists and is already used in the project: `Offset`. As a value class, `Offset` becomes just a `Long` during compilation and there are no allocations (and no class). I changed `Point` to a value class wrapping `Offset`.

This touches many parts of the project, and I am interested to know what you think. I have also updated some of the public API, making this a breaking change, so the release with this would be a major version bump.

Tested everything in the demo app and my app.

### CODE REVIEW

**Comment by peterLaurence** [comment]
`Offset` can hold two `Float`s, but not two `Double`s. The are many places in this library where `Double` is used, and for a reason.
This change, while harmless at first sight, can introduce hard to debug issues.
Although there may be few places, especially in some APIs, where a pair of `Float` would have been enough. For each individual cases, I'm open to discuss and change some API signature. I'm pretty sure there are some good ideas in you PR to keep (especially the `Point` class which is much more descriptive than "Offset"). But I want to keep the core library unaffected.

It's a bit awkward, because you must have worked on it for quite some time. For changes of this magnitude, it's best to have a discussion beforehand. I would have said that right away.

To summarize, if we agree on a list of API signatures to change, I should publish a new version of the library (which would include all the latest fixes), then move on to the 2.x branch. My plan is to have 2.0 include compose 1.1.0.

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt [hunk 1]
```diff
@@ -203,15 +200,14 @@ fun MapState.onCalloutClick(cb: (id: String, x: Double, y: Double) -> Unit) {
  * @param id The id of the marker
  * @param deltaPx The displacement amount in pixels
  */
-fun MapState.moveMarkerBy(id: String, deltaPx: Offset) {
+fun MapState.moveMarkerBy(id: String, deltaPx: Point) {
```
Comment (peterLaurence) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt:203]: In this case, an `Offset` is more descriptive than `Point`.

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt [hunk 2]
```diff
@@ -179,19 +176,19 @@ fun MapState.markerDerivedState(): State<List<MarkerDataSnapshot>> {
...
     return snapshotFlow {
         markerState.markers.values.map {
-            MarkerDataSnapshot(it.id, it.x, it.y)
+            MarkerDataSnapshot(it.id, it.position)
         }
     }
 }
 
-data class MarkerDataSnapshot(val id: String, val x: Double, val y: Double)
+data class MarkerDataSnapshot(val id: String, val position: Point)
```
Comment (peterLaurence) [inline mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt:184]: This is a typical place where `Double` should be used.

File: mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt (hunk 1 above)
Comment (peterLaurence) [reply mapcompose/src/main/java/ovh/plrapps/mapcompose/api/MarkerApi.kt:203]: I'm not reviewing all the changes, because there are too many of them. Just came across these.

**Review by peterLaurence — CHANGES_REQUESTED** [review]

**Comment by peterLaurence** [comment]
For a more precise guidance:
- when the input is relative coordinates, use `Double`
- when the input is a pixel amount, use `Point` or `Offset` depending on the context

**Comment by Nohus** [comment]
> It's a bit awkward, because you must have worked on it for quite some time. For changes of this magnitude, it's best to have a discussion beforehand. I would have said that right away.

Haha it didn't take that long, I thought my proposal would be easiest to discuss when the changes are visible. : )

**Comment by Nohus** [comment]
> Offset can hold two Floats, but not two Doubles. There are many places in this library where Double is used, and for a reason.

That's one thing I suspected, what is the need for Doubles? It didn't look to me like the precision of Floats is not enough in my testing, of course I'm not as familiar with the code as you.

**Comment by peterLaurence** [comment]
Floats precision is not enough when offsetting position. On the top of my head, drag gestures require Double precision. There other cases though.
A rule of thumb is:
Use Float or Int for pixel coordinate. Relative coordinates should have Double precision.

**Comment by Nohus** [comment]
> Use Float or Int for pixel coordinate. Relative coordinates should have Double precision.

What do you think about a `PointF` value class with floats (the current one), and new `Point` class with Doubles (not a value class, so with allocations) for cases that need it?

I think it could be useful to try moving places that require Double to a Double-based `Point`, and then use the profiler to measure exactly how much the allocations cost. If it's too much, the places that require a `Double` _and_ are run every frame could be left with using separate x and y coordinates, while still leaving most of the code with the nicer `Point`-based API.

**Comment by peterLaurence** [comment]
> What do you think about a PointF value class with floats (the current one), and new Point class with Doubles (not a value class, so with allocations) for cases that need it?

Regarding the core library (not APIs), I don't see the benefit of wrapping x and y Double coordinates in a class (besides increasing a bit the pressure on the GC). It's not that costly in term of code lisibility.

After an overview of existing APIs, I noticed a few APIs which would benefit from a `Point` class with Doubles:
- `MapState.visibleArea()`. I have to admit, all those `p1x`, `p2x`.. attributes aren't pretty at all.
This is where a `Point` class with Doubles could be used. However, this is run every frames... In the end, it's a balance between performance and lisibility. 
- `MapState.centroidX` and `MapState.centroidY`. They could be fused in one API, yes!


I had a look at markers, paths, APIs, and I don't think using a wrapper class would bring any valuable improvements.

**Comment by Nohus** [comment]
> I had a look at markers, paths, APIs, and I don't think using a wrapper class would bring any valuable improvements.

Fair enough!
